test/addons-napi/test_promise/test.js

```diff
@@ -24,23 +26,24 @@ const assert = require('assert');
   promise.then(
     common.mustNotCall(),
     common.mustCall(function(result) {
-      assert.strictEqual(result, expected_result,
-                         `promise rejected as expected, received ${result}`);
+      assert.strictEqual(result, expected_result);
     }));
   test_promise.concludeCurrentPromise(expected_result, false);
 }
 
 // Chaining
-const promise = test_promise.createPromise();
-promise.then(
-  common.mustCall(function(result) {
-    assert.strictEqual(result, 'chained answer',
-                       'resolving with a promise chains properly');
-  }),
-  common.mustNotCall());
-test_promise.concludeCurrentPromise(Promise.resolve('chained answer'), true);
+{
+  const expected_result = 'chained answer';
+  const promise = test_promise.createPromise();
+  promise.then(
+    common.mustCall(function(result) {
+      assert.strictEqual(result, expected_result);
+    }),
+    common.mustNotCall());
+  test_promise.concludeCurrentPromise(Promise.resolve('chained answer'), true);
+}
 
-assert.strictEqual(test_promise.isPromise(promise), true);
+assert.strictEqual(test_promise.isPromise(test_promise.createPromise()), true);
```

Comment:
@kkwoker @gabrielschulhof Any idea if this changes the test in a signficant/important way? Specifically, I'm not sure if `promise` having a `.then()` already attached and possibly being resolved are signficant to this assertion or not.

Reply:
@Trott I don't think it's significant.